src/protocol.ts

```diff
@@ -0,0 +1,289 @@
+import { Config } from './config';
+import type { Column, QueryResult, Row, Rows, SQParam } from './skytable';
+
+const PARAMS_TYPE = {
+  NULL: '\x00',
+  BOOLEAN: '\x01',
+  UINT: '\x02',
+  SINT: '\x03',
+  FLOAT: '\x04',
+  BINARY: '\x05',
+  STRING: '\x06',
+};
+
+const RESPONSES_RESULT = {
+  NULL: 0,
+  BOOL: 1,
+  U8INT: 2,
+  U16INT: 3,
+  U32INT: 4,
+  U64INT: 5,
+  S8INT: 6,
+  S16INT: 7,
+  S32INT: 8,
+  S64INT: 9,
+  FLOAT32: 10,
+  FLOAT64: 11,
+  BINARY: 12,
+  STRING: 13,
+  LIST: 14,
+  ERROR: 0x10,
+  ROW: 0x11,
+  EMPTY: 0x12,
+  MULTIROW: 0x13,
+};
+
+const HANDSHAKE_RESULT = {
+  SUCCESS: 'H00',
+  ERROR: 'H01',
+};
+
+function isResponsesResult(type: number): boolean {
+  return Object.values(RESPONSES_RESULT).includes(type);
+}
+
+function isFloat(number: number | string): boolean {
+  return Number.isFinite(number) && !Number.isInteger(number);
+}
+
+export function encodeParams(parameters: SQParam[]): string {
+  if (!parameters.length) {
+    return '';
+  }
+
+  return parameters
+    .map((param) => {
+      switch (typeof param) {
+        case 'string':
+          return [PARAMS_TYPE.STRING, param.length, '\n', param].join('');
+        case 'number':
+          // 2 Unsigned integer 64
+          // 3 Signed integer 64
+          // 4 Float A 64-bit
+          return [
+            isFloat(param)
+              ? PARAMS_TYPE.FLOAT
+              : param < 0
+                ? PARAMS_TYPE.SINT
+                : PARAMS_TYPE.UINT,
+            String(param),
+            '\n',
+          ].join('');
+        case 'bigint':
+          return [
+            param < 0 ? PARAMS_TYPE.SINT : PARAMS_TYPE.UINT,
+            String(param),
+            '\n',
+          ].join('');
+        case 'boolean':
+          return [PARAMS_TYPE.BOOLEAN, Number(param) === 1 ? '\x01' : 0].join(
+            '',
+          );
+        default:
```

Comment:
Is this safe to do? If there is a custom type we can't assume it to be a binary blob